File: bench/python/04-classes-traits/01-point-drawable.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def magnitude(self):
        return (self.x * self.x + self.y * self.y) ** 0.5

    def draw(self):
        print(f"point {self.x:g} {self.y:g}")

point = Point(3.0, 4.0)
point.draw()
print(f"{point.magnitude():g}")

File: bench/rust/04-classes-traits/01-point-drawable.rs
trait Drawable { fn draw(&self); }
struct Point { x: f64, y: f64 }
impl Point { fn magnitude(&self) -> f64 { (self.x * self.x + self.y * self.y).sqrt() } }
impl Drawable for Point { fn draw(&self) { println!("point {} {}", self.x, self.y); } }
fn main() { let point = Point { x: 3.0, y: 4.0 }; point.draw(); println!("{}", point.magnitude()); }
